File: Easy/Filter/Hash Table/2367. Number of Arithmetic Triplets.py
def arithmeticTriplets(nums: list[int], diff: int) -> int:
    count = 0
    nums_set = set(nums)
    for num in nums:
        if num + diff in nums_set and num + 2 * diff in nums_set:
            count += 1
    return count


nums = [0, 1, 4, 6, 7, 10]
diff = 3

print(arithmeticTriplets(nums, diff))

File: PHP/Easy/Filters/Hash_Table/2367-Number_of_Arithmetic_Triplets.php
<?php
function arithmeticTriplets($nums, $diff)
{
    $count = 0;
    for ($i = 0; $i < count($nums); $i++) {
        for ($j = $i + 1; $j < count($nums); $j++) {
            if ($nums[$j] - $nums[$i] == $diff) {
                for ($k = $i + 1; $k < count($nums); $k++) {
                    if ($nums[$k] - $nums[$j] == $diff) {
                        $count += 1;
                    }
                }
            }
        }
    }
    return $count;
}


$nums = [0, 1, 4, 6, 7, 10];
$diff = 3;
echo arithmeticTriplets($nums, $diff);
